src/spend_book.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+// # SpendBook and Forced One-Time-Keys
+//
+// The SpendBook logs each spent DBC along with the reissue transaction it was spent in.
```

Comment:
We should discuss about this.

Let's say we have a reissue with 20 inputs and 20 outputs.    iiuc, the spendbook is logging the same transaction for each DBC, ie 40 times.   That's a lot of duplication.

Or maybe not, if data store (merkle reg?) keeps just one copy ot the transaction "value" with 40 pointers to it?

Reply:
As discussed in our HO, likely an area for future optimization, we can leave this as an implementation detail of the spendbook, for sn_dbc we just need to maintain this API of spend key -> transaction.